Panic errors in query_range calls

We use Promxy with VictoriaMetrics as metrics storage. After VictoriaMetrics update to latest version (1.38.1) we encountered problems with query_range calls - Promxy started to crash during execution of such queries. We found following entry in Promxy log:

```
panic: runtime error: index out of range [0] with length 0

goroutine 66626 [running]:
github.com/jacksontj/promxy/pkg/promhttputil.MergeSampleStream(0x0, 0xc001b19480, 0xc001377fe0, 0xc002100710, 0x1, 0x0)
	/home/tjackson/workspace/golang/src/github.com/jacksontj/promxy/pkg/promhttputil/merge.go:189 +0x7b9
github.com/jacksontj/promxy/pkg/promhttputil.MergeValues.func2(0xc001377fe0)
	/home/tjackson/workspace/golang/src/github.com/jacksontj/promxy/pkg/promhttputil/merge.go:150 +0xcb
github.com/jacksontj/promxy/pkg/promhttputil.MergeValues(0x0, 0x270b8a0, 0xc001e3fc40, 0x270b8a0, 0xc001498220, 0x1, 0xc000ef9260, 0x2744ce0, 0xc000455ab0)
	/home/tjackson/workspace/golang/src/github.com/jacksontj/promxy/pkg/promhttputil/merge.go:162 +0x282
github.com/jacksontj/promxy/pkg/promclient.(*MultiAPI).QueryRange(0xc000997720, 0x2733f60, 0xc001111580, 0xc000268a00, 0x32, 0x0, 0xed6a5ee47, 0x3940c20, 0x0, 0xed6a5ee83, ...)
	/home/tjackson/workspace/golang/src/github.com/jacksontj/promxy/pkg/promclient/multi_api.go:410 +0x7d8
github.com/jacksontj/promxy/pkg/promclient.(*TimeTruncate).QueryRange(0xc0003487e0, 0x2733f60, 0xc001111580, 0xc000268a00, 0x32, 0x0, 0xed6a5ee47, 0x3940c20, 0x0, 0xed6a5ee83, ...)
	/home/tjackson/workspace/golang/src/github.com/jacksontj/promxy/pkg/promclient/time_truncate.go:34 +0x1cd
```

These crashes occur every few seconds. Luckily we started VictoriaMetrics update from our dev environment so we were able to discover it before going to production :) We use Promxy 0.0.57 but we also tried the latest version in our setup - crashes still occur. We also tried to inspect failing queries - unfortunately crashes looks nondeterministic. We tried to execute failing queries manually and we were not able to reproduce restarts.

Example of one of failing queries:
```
/api/v1/query_range?query=sum(process_descriptors_usage{service="test-service"}) by (instance, port)&end=1595148309&start=1595148249&step=60s
```

--- a/pkg/promhttputil/merge.go
+++ b/pkg/promhttputil/merge.go
@@ -181,6 +181,13 @@ func MergeSampleStream(antiAffinityBuffer model.Time, a, b *model.SampleStream) 
 		return nil, fmt.Errorf("cannot merge mismatch fingerprints")
 	}
 
+	// if either set of values are empty, return the one with data
+	if len(a.Values) == 0 {
+		return b, nil
+	} else if len(b.Values) == 0 {
+		return a, nil
+	}
+
 	// TODO: really there should be a library method for this in prometheus IMO
 	// At this point we have 2 sorted lists of datapoints which we need to merge
 	newValues := make([]model.SamplePair, 0, len(a.Values))

--- a/pkg/promhttputil/merge_test.go
+++ b/pkg/promhttputil/merge_test.go
@@ -509,6 +509,64 @@ func TestMergeValues(t *testing.T) {
 			}),
 			antiAffinity: model.Time(2),
 		},
+
+		{
+			name: "Matrix merge empty A",
+			a: model.Matrix([]*model.SampleStream{
+				{
+					model.Metric(model.LabelSet{model.MetricNameLabel: model.LabelValue("hosta")}),
+					[]model.SamplePair{},
+				},
+			}),
+			b: model.Matrix([]*model.SampleStream{
+				{
+					model.Metric(model.LabelSet{model.MetricNameLabel: model.LabelValue("hosta")}),
+					[]model.SamplePair{{
+						model.Time(100),
+						model.SampleValue(10),
+					}},
+				},
+			}),
+			r: model.Matrix([]*model.SampleStream{
+				{
+					model.Metric(model.LabelSet{model.MetricNameLabel: model.LabelValue("hosta")}),
+					[]model.SamplePair{{
+						model.Time(100),
+						model.SampleValue(10),
+					}},
+				},
+			}),
+			antiAffinity: model.Time(2),
+		},
+
+		{
+			name: "Matrix merge empty A",
+			a: model.Matrix([]*model.SampleStream{
+				{
+					model.Metric(model.LabelSet{model.MetricNameLabel: model.LabelValue("hosta")}),
+					[]model.SamplePair{{
+						model.Time(100),
+						model.SampleValue(10),
+					}},
+				},
+			}),
+			b: model.Matrix([]*model.SampleStream{
+				{
+					model.Metric(model.LabelSet{model.MetricNameLabel: model.LabelValue("hosta")}),
+					[]model.SamplePair{},
+				},
+			}),
+			r: model.Matrix([]*model.SampleStream{
+				{
+					model.Metric(model.LabelSet{model.MetricNameLabel: model.LabelValue("hosta")}),
+					[]model.SamplePair{{
+						model.Time(100),
+						model.SampleValue(10),
+					}},
+				},
+			}),
+			antiAffinity: model.Time(2),
+		},
 	}
 
 	for _, test := range tests {

